Jenkinsfile

```
@@ -48,26 +47,42 @@ jobWrapper {
             targetSHA1 = 'NONE'
             if (isPullRequest) {
                 targetSHA1 = sh(returnStdout: true, script: "git fetch origin && git merge-base origin/${targetBranch} HEAD").trim()
+            } 
+
+            isCoreCronJob = isCronJob()
+            requireNightlyBuild = false
+            if(isCoreCronJob) {
+                requireNightlyBuild = isNightlyBuildNeeded()
             }
-        }
+        }   
 
         currentBranch = env.BRANCH_NAME
         println "Building branch: ${currentBranch}"
-        println "Target branch: ${targetBranch}"
-
+        println "Target branch: ${targetBranch}"        
         releaseTesting = targetBranch.contains('release')
         isMaster = currentBranch.contains('master')
         longRunningTests = isMaster || currentBranch.contains('next-major')
         isPublishingRun = false
         if (gitTag) {
             isPublishingRun = currentBranch.contains('release')
         }
-
+        else if(isCoreCronJob && requireNightlyBuild) {
+            isPublishingRun = true
+            longRunningTests = true
+        }
+        
         echo "Pull request: ${isPullRequest ? 'yes' : 'no'}"
         echo "Release Run: ${releaseTesting ? 'yes' : 'no'}"
         echo "Publishing Run: ${isPublishingRun ? 'yes' : 'no'}"
+        echo "Is Realm cron job: ${isCoreCronJob ? 'yes' : 'no'}"
+        echo "Requires nighly build: ${requireNightlyBuild ? 'yes' : 'no'}"
         echo "Long running test: ${longRunningTests ? 'yes' : 'no'}"
 
+        if(!requireNightlyBuild) {
```

```suggestion
        if (isCoreCronJob && !requireNightlyBuild) {
```
I missed this on the first pass, but if I understand correctly, we want to abort the build only when it was triggered by the cron job, but there weren't any new commits.

---

`requireNightlyBuild` is supposed to be `false` if `isCoreCronJob` is `false`. So this should happen either way. Anyway, better to have an explicit `if`